plugins/chat/app/controllers/api/chat_channels_controller.rb

```diff
@@ -38,6 +38,13 @@ def destroy
 
     begin
       ChatChannel.transaction do
+        channel_from_params.update!(
+          slug:
+            "#{Time.now.strftime("%Y%m%d-%H%M")}-#{channel_from_params.slug}-deleted".truncate(
+              SiteSetting.max_topic_title_length,
+              omission: "",
+            ),
+        )
```

Comment:
Note this is an ongoing reflexion of @martin-brennan and I to have proper service object and other patterns for controllers. This is not the case yet though and for now I prefer to have all the logic here even if not ideal.